src/util.cpp

```diff
@@ -1218,609 +1218,34 @@ QString CLocale::GetCountryFlagIconsResourceReference ( const QLocale::Country e
-            strISO3166 = "sa";
-            break;
-        case 187:
-            strISO3166 = "sn";
-            break;
-        case 188:
-            strISO3166 = "sc";
-            break;
-        case 189:
-            strISO3166 = "sl";
-            break;
-        case 190:
-            strISO3166 = "sg";
-            break;
-        case 191:
-            strISO3166 = "sk";
-            break;
-        case 192:
-            strISO3166 = "si";
-            break;
-        case 194:
-            strISO3166 = "so";
-            break;
-        case 195:
-            strISO3166 = "za";
-            break;
-        case 197:
-            strISO3166 = "es";
-            break;
-        case 198:
-            strISO3166 = "lk";
-            break;
-        case 201:
-            strISO3166 = "sd";
-            break;
-        case 202:
-            strISO3166 = "sr";
-            break;
-        case 204:
-            strISO3166 = "sz";
-            break;
-        case 205:
-            strISO3166 = "se";
-            break;
-        case 206:
-            strISO3166 = "ch";
-            break;
-        case 207:
-            strISO3166 = "sy";
-            break;
-        case 208:
-            strISO3166 = "tw";
-            break;
-        case 209:
-            strISO3166 = "tj";
-            break;
-        case 210:
-            strISO3166 = "tz";
-            break;
-        case 211:
-            strISO3166 = "th";
-            break;
-        case 212:
-            strISO3166 = "tg";
-            break;
-        case 214:
-            strISO3166 = "to";
-            break;
-        case 216:
-            strISO3166 = "tn";
-            break;
-        case 217:
-            strISO3166 = "tr";
-            break;
-        case 221:
-            strISO3166 = "ug";
-            break;
-        case 222:
-            strISO3166 = "ua";
-            break;
-        case 223:
-            strISO3166 = "ae";
-            break;
-        case 224:
-            strISO3166 = "gb";
-            break;
-        case 225:
-            strISO3166 = "us";
-            break;
-        case 227:
-            strISO3166 = "uy";
-            break;
-        case 228:
-            strISO3166 = "uz";
-            break;
-        case 231:
-            strISO3166 = "ve";
-            break;
-        case 232:
-            strISO3166 = "vn";
-            break;
-        case 236:
-            strISO3166 = "eh";
-            break;
-        case 237:
-            strISO3166 = "ye";
-            break;
-        case 239:
-            strISO3166 = "zm";
-            break;
-        case 240:
-            strISO3166 = "zw";
-            break;
-        case 242:
-            strISO3166 = "me";
-            break;
-        case 243:
-            strISO3166 = "rs";
-            break;
-        case 248:
-            strISO3166 = "ax";
-            break;
-        }
-        strReturn = ":/png/flags/res/flags/" + strISO3166 + ".png";
+        return "";
```

Comment:
`":/png/flags/res/flags/flagnone.png";`?  Or is the caller using `""` for something special?

Reply:
So the called wants:
* either a valid pixmap for use as a flag icon
* or null, to indcate no flag icon could be found

Changing null to `""`, which is not null, will change behaviour.

Reply:
> Changing null to "", which is not null, will change behaviour.

The `return ""`s we talk about are within `CLocale::GetCountryFlagIconsResourceReference`. The `IsNull()` checks we talk about are on `QPixmap` instances which got passed the result of `GetCountryFlagIconsResourceReference` (i.e. `""`). According to the [docs](https://doc.qt.io/qt-5/qpixmap.html#QPixmap-2), a freshly constructed `QPixmap` whose fileName could not be loaded becomes a null pixmap, so I'd assume that this works as expected.
> If the file does not exist or is of an unknown format, the pixmap becomes a null pixmap.

The existing flag lookup code returns an empty string as well, so I don't expect any differences there.
https://github.com/jamulussoftware/jamulus/blob/5f8ce856114f3ff34ba7bbd43bfaac0ff9579b52/src/util.cpp#L1785-L1788

Anyway, I'm still planning to add (temporary) tests which show the behavior of the old code vs. the new code so that any differences can be spotted and fixed if necessary.

Reply:
> The existing flag lookup code returns an empty string as well, so I don't expect any differences there.

I've now added the tests to the PR description and I don't see any negative changes in behavior due to this (or other parts).

Reply:
Thanks - I think I lost track of which value was being returned to which method...